labs/b8.md

```diff
@@ -0,0 +1,84 @@
+2. How does one import a key?
+3. How does one export a key to a file? (Hint: use `--armor`)
+4. How does one see all of the keys on his/her keyring?
+5. Use this private key `privkey` to decrypt `file2.txt.gpg` (*for real-life purposes, please actually keep your private keys secret*). What are the decrypted contents of `file2.txt.gpg`?
+
+## Hashing (Checksums)
+Hash functions deterministically map arbitrary-length data to a fixed-length hash (in other words, a fixed number of bytes). As a result, the latter can serve as a summary of the former if the former exceeds the latter in terms of byte length. For instance, if we download a 1GB file and want to verify its integrity, instead of re-downloading the entire file again, we can simply compute a 20B hash of the file on our end and compare it to the 20B hash of source.
+
+To try it out:
+1. `sha1sum [FILE]` to get the SHA1 hash of `[FILE]`.
+2. `md5sum [FILE]` to get the MD5 hash of `[FILE]`.
+
+There are [many hash functions](https://en.wikipedia.org/wiki/List_of_hash_functions), some of which are cryptographic hash functions and others not. Crytographic hash functions primarily differ from non-cryptographic hash functions in that they are difficult to forge. For instance, if someone has a hash on hand, it is difficult for them to create a pre-hash file that maps to the same hash. If you are interested in all of the properties of a cryptographic hash function,  [read here](https://en.wikipedia.org/wiki/Cryptographic_hash_function).
+
+### Lab Activity
+1. What is the MD5 hash of `file3.txt`?
+2. What is the SHA1 hash of the MD5 hash of `file3.txt`? In other words, what is `SHA1(MD5(file3.txt))`?
+
+## File Security
+The UNIX permission model has 3 components: permissions given to the file's (1) owning user, (2) owning group and (3) others/everyone else. Permissions themselves have 3 subcomponents: (1) read, (2) write and (3) execute, which are self-explanatory in respectively enforcing the ability to read, write or execute a file.
+
+![alt](http://www.comentum.com/images/permissions.jpg)
+
+To try it out:
+1. `ls -l` to view the permissions of all files in the current directory.
+2. `chown [NEWUSER]:[NEWGROUP] [FILE]` to change `[FILE]`'s  user and group ownerships respectively to `[NEWUSER]` and `[NEWGROUP]`.
+3. `chmod [PERMISSIONS] [FILE]` to set `[FILE]` with specified `[PERMISSIONS]`.
+
+### Lab Activity
+1. `file4.txt`: What are the permissions of this file? Explain what they allow and disallow.
+2. `file5`: Make this file executable and execute it. What is its printout?
+3. `file6.txt`: Change this file to be under your ownership.
+4. `file7.txt`: Make this file readable only to you. 
+5. `file8.txt`: Change this file's permissions such that only root should be able to read this file and no one should be able to edit it.
+6. `file9.txt`: These files are currently owned by another user. Choose any method to make the files readable to you and unreadable to the previous owner.
```

Comment:
Link the Google Form for submitting the lab around here somewhere? (probably linked both at the start and end of the questions so that people can easily find it)